src/rt.rs

```diff
@@ -20,16 +20,16 @@ pub trait Executor<Fut> {
 /// A timer which provides timer-like functions.
 pub trait Timer {
```

Comment:
Unrelated: Would it be good to have the resulting futures as an associated type rather than as a boxed future? Both `tokio` and `smol` have well-defined "sleep" types instead of `async` methods, so we may be able to avoid an allocation here by not boxing it.

Reply:
That would require parameterizing the ~entire API over the timer trait which I'm not sure is desired.

Reply:
As a workaround, you could just do:

```rust
struct BoxedTimer {
    inner: Box<dyn Timer<Sleep = Pin<Box<dyn Future>>>>>
}
```

But this might not be idiomatic, so I agree with your assessment.